plugins/hls-explicit-record-fields-plugin/src/Ide/Plugin/ExplicitFields.hs

```diff
@@ -347,14 +354,19 @@ collectRecords = everything (<>) (maybeToList . (Nothing `mkQ` getRecPatterns `e
 collectNames :: GenericQ (UniqFM Name [Name])
 collectNames = everything (plusUFM_C (<>)) (emptyUFM `mkQ` (\x -> unitUFM x [x]))
 
-getRecCons :: LHsExpr (GhcPass 'Renamed) -> Maybe RecordInfo
+getRecCons :: LHsExpr (GhcPass 'Renamed) -> (Maybe RecordInfo, Bool)
+-- When we stumble upon an occurrence of HsExpanded, we only want to follow a
+-- single branch. We do this here, by explicitly returning occurrences from
+-- traversing the original branch, and returning True, which keeps syb from
+-- implicitly continuing to traverse.
+getRecCons (unLoc -> XExpr (HsExpanded _ expanded)) = (listToMaybe (collectRecords expanded), True)
 getRecCons e@(unLoc -> RecordCon _ _ flds)
-  | isJust (rec_dotdot flds) = mkRecInfo e
+  | isJust (rec_dotdot flds) = (mkRecInfo e, False)
   where
     mkRecInfo :: LHsExpr (GhcPass 'Renamed) -> Maybe RecordInfo
     mkRecInfo expr = listToMaybe
       [ RecordInfoCon realSpan' (unLoc expr) | RealSrcSpan realSpan' _ <- [ getLoc expr ]]
-getRecCons _ = Nothing
+getRecCons _ = (Nothing, False)
 
 getRecPatterns :: LPat (GhcPass 'Renamed) -> Maybe RecordInfo
 getRecPatterns conPat@(conPatDetails . unLoc -> Just (RecCon flds))
```

Comment:
ConPat can also occur inside an HsExpanded expression. For example, the constructed example above uses ConPat (because the wildcards occur in the let binding)

Reply:
Actually thinking about it though, you shouldn't need to change this, because `getRecCon` should already, by calling `collectRecords`, return both `RecordCon` and `ConPat` matches. May want to point that out in the comments though.